crates/cargo-plumbing-schemas/src/read_manifest.rs

```diff
@@ -0,0 +1,30 @@
+use std::{io::Read, marker::PhantomData};
+
+pub use cargo_util_schemas::manifest::TomlManifest;
+use serde::{Deserialize, Serialize};
+
+use crate::MessageIter;
+
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "reason", rename_all = "kebab-case")]
+#[cfg_attr(feature = "unstable-schema", derive(schemars::JsonSchema))]
+pub enum ReadManifestMessage {
+    Manifest { manifest: TomlManifest },
```

Comment:
This needs some kind of identifier to distinguish the different manifests.

Connecting the message with the input will require a `Path`

Connecting the message with future operations may require a `PackageIdSpec`

Reply:
Let's go with absolute path.